ts(2307) for importing Vue files from other nx libraries

### Current Behavior

Cannot find module '@vue-ts-test/ui-components' or its corresponding type declarations. ts(2307)

### Expected Behavior

no error

### GitHub Repo

https://github.com/DDuunk/vue-ts-test

### Steps to Reproduce

1. Create a integrated monorepo using vue
2. pnx g @nx/vue:lib ui-components --directory=libs/ui/components --style=scss
3. pnx g @nx/vue:lib ui-layouts --directory=libs/ui/layouts --style=scss
4. Create a vue3 file in ui-component, lets call it `component-a`
5. Create a vue3 file in ui-layouts, lets call it `layout-a`
6. Import `component-a` in `layout-a`


### Nx Report

```shell
Node   : 18.18.0
   OS     : darwin-arm64
   npm    : 9.8.1
   
   nx (global)        : 16.2.2
   nx                 : 17.2.8
   @nx/js             : 17.2.8
   @nx/linter         : 17.2.8
   @nx/eslint         : 17.2.8
   @nx/workspace      : 17.2.8
   @nx/devkit         : 17.2.8
   @nx/eslint-plugin  : 17.2.8
   @nrwl/tao          : 17.2.8
   @nx/vite           : 17.2.8
   @nx/vue            : 17.2.8
   @nx/web            : 17.2.8
   typescript         : 5.2.2
```


### Failure Logs

_No response_

### Package Manager Version

pnpm 8.9.0

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/vue/src/generators/library/files/tsconfig.lib.json__tmpl__
+++ b/packages/vue/src/generators/library/files/tsconfig.lib.json__tmpl__
@@ -2,16 +2,7 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "outDir": "<%= offsetFromRoot %>dist/out-tsc",
-    "baseUrl": ".",
-    "paths": {
-      "@/*": ["./src/*"]
-    },
     "types": [],
-    "lib": [
-      "ES2016",
-      "DOM",
-      "DOM.Iterable"
-      ],
   },
   "exclude": [
     "src/**/__tests__/*", 
@@ -22,7 +13,9 @@
     "src/**/*.spec.js", 
     "src/**/*.test.js", 
     "src/**/*.spec.jsx", 
-    "src/**/*.test.jsx"
+    "src/**/*.test.jsx",
+    "src/**/*.spec.vue",
+    "src/**/*.test.vue"
     ],
   "include": [
     "src/**/*.js", 

--- a/packages/vue/src/generators/library/library.spec.ts
+++ b/packages/vue/src/generators/library/library.spec.ts
@@ -78,6 +78,7 @@ describe('lib', () => {
   it('should add vue, vite and vitest to package.json', async () => {
     await libraryGenerator(tree, defaultSchema);
     expect(readJson(tree, '/package.json')).toMatchSnapshot();
+    expect(tree.read('my-lib/tsconfig.lib.json', 'utf-8')).toMatchSnapshot();
   });
 
   it('should update root tsconfig.base.json', async () => {

